examples/tv-app/android/java/MyUserPrompter-JNI.cpp

```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
 void JNIMyUserPrompter::PromptWithCommissionerPasscode(uint16_t vendorId, uint16_t productId, const char * commissioneeName,
                                                        uint32_t passcode, uint16_t pairingHint, const char * pairingInstruction)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+    std::string stringPairingInstruction(pairingInstruction);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```

Comment:
same as the above, use env->NewStringUTF, thanks

Reply:
addressed